### DISCUSSION TITLE
Welche Einstellung für "Batteriewerte aus MQTT-Broker"

### DISCUSSION THREAD

**Discussion by pse66** [hoylabs/OpenDTU-OnBattery#929] [General]
Hallo in die Runde.

Ich bin neu in der OpenDTU-onBattery Thematik und habe hier viel gelesen, gelernt und konnte mein Projekt soweit umsetzen. Danke für die Super-Software!

Ich habe aber eine Frage bzgl. der „Batterie Funktion“ über MQTT.

Ich betreibe einen HM1500 an einer Batterie die mit einem JK BMS ausgestattet ist als „Nulleinspeisung“. Die Batterie-Daten werden vom JK BMS per MQTT an meinen ioBroker gesendet. 
OpenDTU-onBattery ist ebenfalls per MQTT mit dem ioBroker verbunden.

Was muss ich hier bei OpenDTU-onBattery eintragen, damit die Batteriewerte wie SOC und Spannung korrekt vom ioBroker übernommen werden, bzw. funktioniert das?

![Clipboard01](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/168324651/20ef767d-f93f-40dd-853f-f8b1961e56e8)

Viele Grüße
Frank

**Comment by schlimmchen** [comment]
Wenn du ein JK BMS hast, verstehe ich nicht ganz, warum du es nicht seriell an OpenDTU-OnBattery anschließt. Aber das ist dir natürlich freigestellt.

Ansonsten deutet deine Frage an, dass du MQTT noch nicht ganz durchdrungen hast. Du musst hier lediglich eintippen, an welchem Topic die SoC bzw. die Batteriespannung am MQTT Broker veröffentlicht werden. Wo (an welchem Topic) die veröffentlicht werden, muss du selbst wissen.

Was bekommt denn die Daten vom JK BMS und schickt die an den MQTT Broker? Können die "new style" JK BMS das von Haus aus? Haben die WLAN?

**Reply by pse66** [reply]
ich hole die Daten per ESP über BT und sende sie per MQTT zum iobroker, da ich die Daten noch für andere Geräte brauche. 
Schreibweise beim Topic war falsch. Jetzt passt es und die Daten landen im OpenDTU-OnBattery!

**Comment by peff74** [comment]
Oh, bei der Frage hänge ich mich gleich mal mit rein.
Ich ziehe, meine Daten von meinem JK BMS per Bluetooth, da Kabelwege zu umständlich wären und ich parallel auch gleich meine Akkuheizung steuere.
Habe nun auch gesehen, dass man die Werte über Batterie/MQTT in die OpenDTU bekommt.
Ich habe jetzt aber nur die Systemspannung und nicht SoC eingetragen.
Wenn ich die Ausgabe der Konsole korrekt deute, wird diese jetzt auch genutzt.

Ist SoC ebenfalls zwingend notwendig?

Bei meinem kleinen JK ist dieser nämlich mehr ein Schätzwert/Spielerei und ich gebe mehr auf die Spannung.
Ich habe jetzt unter DPL den Schalter "Batterie SoC ignorieren" aktiviert.
Somit sollten ja die Start/Stop Schwellwerte gelten, die jetzt ja direkt vom JK-BMS kommen, korrekt?
Nur die Anzeige in der Live Ansicht zeigt nun 0%, was mich nicht stört^^

**Comment by schlimmchen** [comment]
> Ich habe jetzt unter DPL den Schalter "Batterie SoC ignorieren" aktiviert.

Dann musst du auch nicht das SoC Topic abonnieren.

> Somit sollten ja die Start/Stop Schwellwerte gelten, die jetzt ja direkt vom JK-BMS kommen, korrekt?

Wenn die vom JK BMS in den Broker kommen und du das passende Topic abonnierst, dann wird auch diese Spannung im DPL benutzt.

Wie kommen denn die Daten über Bluetooth in einen Broker? Habt ihr einen weiteren Mikrocontroller, der das macht?

**Reply by pse66** [reply]
https://github.com/syssi/esphome-jk-bms

**Reply by peff74** [reply]
Ja, nutze ich auch. Als Standalone ohne HomeAssistant
Auf einem separaten ESP.

**Comment by Manos1966** [comment]
@pse66 Ich erzaehle wie ich das mache:

- Ich habe einen MQTT Server
- Meine Devices senden per MQTT Ihre Daten zum MQTT Server
- Ich weiss aber (am Anfang) nicht, wie sie heissen und welche Daten sie senden

Ich installiere den MQTT Explorer https://mqtt-explorer.com/
Hast du Home Assistant, kannst du ihn direkt im Home Assistant einbinden.
Oder einfach die Windows Anwendung installieren.

Mit MQTT Explorer logs du dich auf deinem MQTT Server ein.
Und dann siehst du Real-Time alle Daten die auf deinem MQTT Server landen!
Dann siehst du die Daten die dich interessieren z.B. coffee_maker (siehe Bild)
clickst du drauf, dann siehst du auf der rechten Seite unter Topic, wie dein MQTT Topic heissen soll, z.B. kitchen/coffee_maker dass ist die Antwort zu deiner Frage.
Ich hoffe es hilft.

![MQTT Explorer screen3](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/8f50d892-e84d-40c8-9991-63f41876b80f)

**Reply by peff74** [reply]
Den MQTT Explorer kann ich ebenfalls nur empfehlen!
Ich nutze den MQTT Broker vom ioBroker, aus der Historie heraus, weil ich damit mal angefangen habe.
Was ein nettes Feature ist, am ioBroker man kann relativ leicht Reaktionen auf Werte erzeugen, Schalte dies / schicke Benachrichtigung / etc.

**Comment by JanStroeder** [comment]
Ich nutze einen Victron Cerbo als MQTT Broker, worin die Batteriewerte auch gepublished werden. Die Batterie aktualisiert die Werte allerdigns nur in ziemlichen langen Intervallen (1h und groesser). Daher habe ich mal in Node-Red die Werte in einen custom Topic gepublished und retained gestellt. Zusammen mit der Option in Open-DTU on battery  > Batterie Einstellungen > JSON-Pfad: "value" eingetragen. Damit funktioniert es. 
Ich vermute nach den vorliegenden Informationen, dass das eigentlich nicht notwendig sein dürfte, weil der "value" aus dem Topic ja wie gewünscht einen "numerischen Wert" enthält.
<img width="1160" height="288" alt="grafik" src="https://github.com/user-attachments/assets/beb376f5-5b27-4f9d-91b0-441f22739785" />

**Comment by knopers1** [comment]
Ich habe jetzt auch auf Winterbetrieb umgestellt, dabei den Akku über MQTT angebunden..... Die Werte werden regelmäßig auf dem iObroker aktualisiert. tauchen aber bei openDTUonBattery nur einmallig auf, sobald die Einstellung gespeichert wird. Wieso holt sich die openDTU die Werte nicht regelmäßig ab? 

<img width="819" height="649" alt="1" src="https://github.com/user-attachments/assets/eebcb71e-240f-4918-a65c-697a5e1f3713" />
<img width="1577" height="195" alt="2" src="https://github.com/user-attachments/assets/625e9b76-9d65-4622-ab77-d8a02b832485" />
<img width="1395" height="420" alt="3" src="https://github.com/user-attachments/assets/678686d4-25ad-482b-9a3e-4099397874d6" />

**Reply by knopers1** [reply]
Firmwareversion 2025.06.11, jetzt auf die 2025.10.11 gewechselt, dort exakt das selbe Verhalten.

**Reply by gitisgreat2023** [reply]
Schau mal im logging ob er einen fehlgeschlagene MQTT refresh macht.

**Reply by knopers1** [reply]
habe ein Log-Auszug gemacht. Der MQTT taucht nur beim "Batterie Einstellung" -> Aktiviere Schnittstelle -> Speichern -> im Log auf...

[06:08:31.143] D (220211356) mqtt: Received MQTT message on topic 'openDTUWinter/battery/stateOfCharge' (Bytes 1-2/2)
[06:08:31.143] D (220211356) battery: [MQTT] Updated SoC to 71 from 'openDTUWinter/battery/stateOfCharge'
[06:08:31.143] D (220211357) mqtt: Received MQTT message on topic 'openDTUWinter/battery/voltage' (Bytes 1-5/5)
[06:08:31.143] D (220211357) battery: [MQTT] Updated voltage to 45.88 from 'openDTUWinter/battery/voltage'
[06:08:31.143] D (220211357) mqtt: Received MQTT message on topic 'openDTUWinter/battery/current' (Bytes 1-1/1)
[06:08:31.239] D (220211357) battery: [MQTT] Updated current to 0 from 'openDTUWinter/battery/current'

Danach kommt dies in regelmäßigen Intervallen:

[06:08:44.605] D (220225055) dynamicPowerLimiter: [Controller] battery interface enabled, SoC 71.0 % (used), age 13 s (valid)
[06:08:49.422] D (220230203) dynamicPowerLimiter: [Controller] battery interface enabled, SoC 71.0 % (used), age 18 s (valid)


Der Wert SoC 71.0% bleibt die ganze Zeit stehen, obwohl sich mit der Zeit der SOC Wert auf 72,73% auf dem Akku  ändert....  Dies kann ich auch auf dem IObroker beobachten, dort scheint die Kommunikation mit dem Akku über MQTT sauber zu funktionieren, da die drei Werte alle 5 sek neu geschrieben werden.... Die DTU holt sich die Werte aus welchem Grund auch immer nicht ab!

![1](https://github.com/user-attachments/assets/99dbbf10-f443-4655-b078-6983c7679008)
![2](https://github.com/user-attachments/assets/31464d3f-d695-4814-8dd0-bbc85af99e97)



[opendtu-log-2025-11-25_06-08-50.txt](https://github.com/user-attachments/files/23738655/opendtu-log-2025-11-25_06-08-50.txt)

**Comment by peff74** [comment]
Das hängt nicht an der FW. Dein problem wird wahrscheinlich am ioBroker liegen, habe da auch schon oft geflucht.
Wie kommt der Wert rein in Deine Objektliste?
Ich vermute, ein Flag wird nicht korrekt gesetzt, wodurch die OpenDTU nicht mitbekommt, dass der Wert neu ist.
Bei mir lag es daran dass in meinem Skript **steuere** und nicht **beschreibe** das Objekt genutzt werden musste.

**Reply by knopers1** [reply]
Der Wert kommt von der openDTUonBattery!( openDTUWinter) Dort ist nur mein JK-BMS angeschlossen. Über mqtt landen die Werte des Akkus in die Objektliste. Diese Werte habe ich jetzt auf die openDTUonBattery über MQTT als Akku konfiguriert.

Bis jetzt hatte ich keine Probleme mit mqtt am IObroker. 
Zum Versändniss: 
Ich habe zwei openDTUonBattery im Betrieb. An dem openDTUWinter ist nur der Akku dran, an dem openDTUonBattery ist der Rest angeschlossen.... spricht Akku über + und -, Hoymiles, Victron 150/35, Huawei....

Grundsetzlich kommen die Daten über mqtt von der openDTUWinter regeläßig rein, sie werden alle paar Sekunden aktualisiert. Diese Werte werden leider nur beim Speichern "Batterie Einstellung" auf der openDTUonBattery einmalig übernommen. 

Welcher Flag ist dafür im IoBroker (mqtt) verantwortlich?

**Reply by gitisgreat2023** [reply]
Probiere mal aus den Wert zu ändern in IOBroker, in der object explorer.
Einfach auf den Wert einmal klicken. Und mal ohne und mit "acknowledged" wählen.
Und schauen ob die DTU die Änderung sieht.

<img width="732" height="451" alt="image" src="https://github.com/user-attachments/assets/d7432b69-24d4-47f7-a05b-5ae378841a23" />

**Reply by knopers1** [reply]
Bingo,
ohne "acknowledged" den Wert gesendet, und der Wert wird in der DTU einmalig aktualisiert! 
![1](https://github.com/user-attachments/assets/52912f33-250d-46b9-b1cd-55c1e1e6150d)

 Woran kann es liegen?
![1](https://github.com/user-attachments/assets/cb084406-0941-402f-98f0-2b21d2dc1b1d)

**Reply by knopers1** [reply]
Das wars.....
![1](https://github.com/user-attachments/assets/32310a23-6c4c-4b61-b9e5-e65023368921)

Die Einstellung "Sende auch Zustände ack=true" war dafür verantwortlich. Der Haken muss da sein!
Jetzt aktualisiert sich auch der Akku über mqtt in der DTU-WebUi.

![1](https://github.com/user-attachments/assets/9577341f-1412-4d68-9f41-07208389b8aa)

@gitisgreat2023 -Besten Dank für deine Hilfe!

**Comment by norbertgase-rgb** [comment]
Hallo in die Runde. 
Ich habe den Victron Laderegler über Mqtt Bridge in mosquitto erfolgreich eingebunden.
Die topics aus dem Screenshot sind auch korrekt wie ich im mqtt Explorer sehen konnte.
Was ich aber nicht herausfinden konnte-was muss hier in openDTUonBattery als JSON-Pfad eingetragen werden? (Victron sendet ja kein einfachen numerischen MQTT topic-Values). 
Danke-wäre mir eine große Hilfe.

mfg 
![IMG_3093](https://github.com/user-attachments/assets/35876417-45a5-48aa-a836-7aa609318d1f)

**Reply by norbertgase-rgb** [reply]
Hab's mir grad selbst beantwortet, aber falls es jemand anderem hilft. Im Feld "JSON-Pfad" schlicht "value" eintragen (der Parameter im Payload in dem Fall bei Victron Gerätschaften. Nun wird mir alles angezeigt (ja ich weiß, dass der SOC nicht korrekt ist-Victron-"Krankheit"):
<img width="1323" height="934" alt="vic-1" src="https://github.com/user-attachments/assets/cf3f069e-6439-4ffd-a23e-41dc6e8207b8" />
<img width="1893" height="390" alt="vic-2" src="https://github.com/user-attachments/assets/a89bae6e-1e6a-4e2a-acfe-56e32dd36b36" />

**Reply by SW-Niko** [reply]
>ja ich weiß, dass der SOC nicht korrekt ist-Victron-"Krankheit"

Die "Krankheit" haben Herstellerübergreifend alle berechneten SoC-Werte. 😞 
Ich verwende einen Victron Smart-Shunt und die ... kann man super an die Batterie anpassen.

**Comment by norbertgase-rgb** [comment]
Ich nutze auch den SmartShunt aber nach dem 1. Winter ist der SoC total strubbelig. Wenn Klärchen ☀️ das nächste Mal den Akku vollknallt setze ich mal 100% und gucke was dann an kWh rauskommt.

**Reply by SW-Niko** [reply]
>setze ich mal 100% 

Äh... eigentlich sollte er sich selbstständig auf 100% setzen. Wenn das nicht funktioniert dann passen deine Einstellungen noch nicht.

Aber, ja über den Winter würde mein SoC auch abdriften. 

Wenn du Lust hast, kannst du dir den "Auflade Helfer" mal anschauen. [Fork](https://github.com/SW-Niko/OpenDTU-OnBattery/wiki/%C3%9Cbersicht) 
Der löst das Problem mit dem .. äh... strubbeligen SoC.
Ich bin da noch auf der Suche nach Freiwilligen, die das testen wollen. 😁

**Comment by norbertgase-rgb** [comment]
Ja ok, das mit den 100% Meinte ich eigentlich so.
Habe 25,6V 100Ah Akku von EcoWorthy und das sind die Einstellungen im Shunt:
![IMG_3097](https://github.com/user-attachments/assets/f8a99cbe-5cad-486e-8e74-b0283d35f55d)

**Reply by SW-Niko** [reply]
>Habe 25,6V 100Ah Akku von EcoWorthy und das sind die Einstellungen im Shunt:

Bingo, das ist auch meine Kombination. 👍 

Die Verwendung von 1% Schweifstrom und 10 Minuten Ladezustandserkennung könnte etwas Problematisch sein.
Aber wenn das bis jetzt mit dem automatischen Kalibrieren geklappt, hat dann must du nichts verändern.

Ein Strom Schwellwert von 100mA ist aber schon ungünstig, wenn man den SoC genau berechnen will.
Auf der einen Seite beschwerst du dich hier öffentlich mit "Victron-Krankheit" und auf der anderen Seite sagst du "Lieber Victron, ich befehle dir: Ignoriere alles unter 100mA bei der SoC-Berechnung. Mein Inverter entzieht der Batterie im Standby ständig 60mA, aber egal ... so genau brauch ich den SoC Wert nicht".  

Vielleicht solltest du das Handbuch nochmal lesen. 😉

**Reply by norbertgase-rgb** [reply]
Ich habe mich ausführlichst damals vor 2 Jahren informiert und auch etwas rumgespielt. Was wären dann Deiner Ansicht Nach die besten Werte die ich eintragen sollte?

**Reply by SW-Niko** [reply]
>Ich habe mich ausführlichst damals vor 2 Jahren informiert

Ok.....

Aber wenn du mich schon fragst, dann teile ich dir abschließend noch meine Gedanken zu dem Thema mit.
Ich habe die Grenze auf 10mA festgelegt, weil ich denke das es nicht Unklug ist, wenn der Victron den Standby Strom berücksichtigt.
